transit_odp/timetables/tasks.py

```diff
@@ -824,6 +831,163 @@ def task_rerun_timetables_etl_specific_datasets():
+                revision = timetables_dataset.live_revision
+                if revision:
+                    revision_id = revision.id
+                    output_id = revision.id
+                else:
+                    raise PipelineException(
+                        f"Serverless reprocessing - no live revision for dataset id {timetables_dataset.id}"
+                    )
+            elif _id_type == "dataset_revision_id":
+                revision_id = timetables_dataset
+                output_id = timetables_dataset
+                logger.info(
+                    f"Serverless reprocessing - running Timetables ETL pipeline for revision id {timetables_dataset}"
+                )
+
+            try:
+                revision = DatasetRevision.objects.get(
+                    pk=revision_id, dataset__dataset_type=TimetableType
+                )
+                if _s3_file_names_ids_map:
+                    s3_file_name = get_file_name_by_id(
+                        revision_id, _s3_file_names_ids_map
+                    )
+                    if s3_file_name:
+                        revision.upload_file = s3_file_name
+                        revision.save()
+
+            except DatasetRevision.DoesNotExist as exc:
+                message = f"Serverless reprocessing - DatasetRevision {revision_id} does not exist."
+                logger.exception(message, exc_info=True)
+                raise PipelineException(message) from exc
+
+            if revision:
+                task_id = uuid.uuid4()
+                task = DatasetETLTaskResult.objects.create(
+                    revision=revision,
+                    status=DatasetETLTaskResult.STARTED,
+                    task_id=task_id,
+                )
+                try:
+                    step_functions_client = StepFunctionsClientWrapper()
+                    step_function_arn = settings.TIMETABLES_STATE_MACHINE_ARN
+
+                    # revision.schema_violations.all().delete()
+                    # revision.post_schema_violations.all().delete()
+                    # revision.pti_observations.all().delete()
+                    # revision.dqs_report.all().delete()
+                    revision.txc_file_attributes.all().delete()
+                    revision.num_of_lines = None
+                    revision.admin_areas.clear()
+                    revision.localities.clear()
+                    revision.services.all().delete()
+                    revision.service_patterns.all().delete()
+                    revision.save()
+
+                    payload = StepFunctionsReprocessPayload(
+                        s3=S3Payload(object=revision.upload_file.name),
+                        inputDataSource=InputDataSourceEnum.FILE_UPLOAD.value,
+                        datasetRevisionId=revision_id,
+                        datasetType="timetables",
+                        publishDatasetRevision=False,
+                        datasetETLTaskResultId=task_id,
```

Comment:
The parent class StepFunctionsTTPayload for class StepFunctionsReprocessPayload expects an int for task_id while it's set to uuid in this task.